docs/Measuring_Resonances.md

```diff
@@ -0,0 +1,339 @@
+default value for `accel_per_hz` parameter in `[resonance_tester]` section, e.g.
+```
+[resonance_tester]
+accel_chip: adxl345
+accel_per_hz: 50  # default is 75
+probe_points: ...
+```
+
+Generated CSV files show power spectral density of the vibrations depending on the
+frequency. Usually, the charts generated from these CSV files are pretty much
+self-explanatory, with the peaks corresponding to the resonance frequencies:
+
+![Resonances](img/test-resonances-x.png)
+
+The chart above shows the resonances for X axis at approx. 50 Hz, 56 Hz, 63 Hz,
+80 Hz and 104 Hz and one cross-resonance for Y axis at ~ 56 Hz. From this, one
+can derive that a good input shaper config in this case could be `2hump_ei` at
+around `shaper_freq_y = 45` (Hz):
+
+|![2-hump EI shaper](img/2hump_ei_65hz.png)|
+|:--:|
+|Input Shaper response to vibrations, lower is better.|
+
+Note that the smaller resonance at 104 Hz requires less of vibration suppression
+(if at all).
+
+## Input Shaper auto-calibration
+
+Besides manually choosing the appropriate parameters for the input shaper
+feature, it is also possible to run an experimental auto-tuning for the
+input shaper.
+
+In order to attempt to measure the resonance frequencies and automatically
+determine the best parameters for `[input_shaper]`, run the following command
+via Octoprint terminal:
+```
+SHAPER_CALIBRATE
+```
+
+This will test all frequencies in range 5 Hz - 120 Hz and generate
+the csv output (`/tmp/calibration_data_*.csv` by default) for the frequency
+response and the suggested input shapers. You will also get the suggested
+frequencies for each input shaper, as well as which input shaper is recommended
+for your setup, on Octoprint console. For example:
+
+![Resonances](img/calibrate-y.png)
+```
+Fitted shaper 'zv' frequency = 56.7 Hz (vibrations = 23.2%)
+Fitted shaper 'mzv' frequency = 52.9 Hz (vibrations = 10.9%)
+Fitted shaper 'ei' frequency = 62.0 Hz (vibrations = 8.9%)
+Fitted shaper '2hump_ei' frequency = 59.0 Hz (vibrations = 4.9%)
+Fitted shaper '3hump_ei' frequency = 65.0 Hz (vibrations = 3.3%)
+Recommended shaper_type_y = 2hump_ei, shaper_freq_y = 59.0 Hz
+```
+If you agree with the suggested parameters, you can execute `SAVE_CONFIG`
+now to save them and restart the Klipper.
+
+
+If your printer is a bed slinger printer, you will need to repeat the
+measurements twice: measure the resonances of X axis with the accelerometer
+attached to the toolhead and the resonances of Y axis - to the bed (the usual
+bed slinger setup). In this case, you can specify the axis you want to run the
+test for (by default the test is performed for both axes):
+```
+SHAPER_CALIBRATE AXIS=Y
+```
+
+You can execute `SAVE_CONFIG` twice - after calibrating each axis.
+
+However, if you connected two accelerometers simultaneously and configured them
+in the following manner:
+```
+[adxl345 adxl345_x]
+cs_pin: ...
+
+[adxl345 adxl345_y]
+cs_pin: ...
```

Comment:
That's good to know! Maybe then I can mention exactly that: 2 chips can be used, as long as they are wired to different boards (or different SPI interfaces)? Or do you think it might be better to skip this feature for now?

Reply:
FWIW, I think the feature is fine.  Just wanted to let you know about that limitation of the adxl345 chips.

Reply:
I explicitly wrote about this limitation in the docs now.